sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiSinkDefaultValuesIT.java

```diff
@@ -0,0 +1,254 @@
+      throws IOException, InterruptedException {
+    TableSchema bqSchema;
+    if (defaultFieldsRequired) {
+      bqSchema =
+          new TableSchema()
+              .setFields(
+                  ImmutableList.of(
+                      new TableFieldSchema().setName("id").setType("STRING"),
+                      new TableFieldSchema().setName("key2").setType("STRING"),
+                      new TableFieldSchema().setName("value").setType("STRING"),
+                      new TableFieldSchema()
+                          .setName("defaultliteral")
+                          .setType("INT64")
+                          .setDefaultValueExpression("42")
+                          .setMode("REQUIRED"),
+                      new TableFieldSchema()
+                          .setName("defaulttime")
+                          .setType("TIMESTAMP")
+                          .setDefaultValueExpression("CURRENT_TIMESTAMP()")
+                          .setMode("REQUIRED")));
+    } else {
+      bqSchema =
+          new TableSchema()
+              .setFields(
+                  ImmutableList.of(
+                      new TableFieldSchema().setName("id").setType("STRING"),
+                      new TableFieldSchema().setName("key2").setType("STRING"),
+                      new TableFieldSchema().setName("value").setType("STRING"),
+                      new TableFieldSchema()
+                          .setName("defaultliteral")
+                          .setType("INT64")
+                          .setDefaultValueExpression("42"),
+                      new TableFieldSchema()
+                          .setName("defaulttime")
+                          .setType("TIMESTAMP")
+                          .setDefaultValueExpression("CURRENT_TIMESTAMP()")));
+    }
+
+    TableSchema sinkSchema = bqSchema;
+    if (!sinkKnowsDefaultFields) {
+      sinkSchema =
+          new TableSchema()
+              .setFields(
+                  bqSchema.getFields().stream()
+                      .filter(tfs -> tfs.getDefaultValueExpression() == null)
+                      .collect(Collectors.toList()));
+    }
+    final TableRow row1 =
+        new TableRow()
+            .set("id", "row1")
+            .set("key2", "bar0")
+            .set("value", "1")
+            .set("defaultliteral", 12);
+    final TableRow row2 = new TableRow().set("id", "row2").set("key2", "bar1").set("value", "1");
+    final TableRow row3 = new TableRow().set("id", "row3").set("key2", "bar2").set("value", "2");
+
+    List<TableRow> tableRows = Lists.newArrayList(row1, row2, row3);
+
+    String tableSpec = createAndGetTablespec(bqSchema);
+    Pipeline p = Pipeline.create();
+
+    BigQueryIO.Write<TableRow> write =
+        BigQueryIO.writeTableRows()
+            .to(tableSpec)
+            .withSchema(sinkSchema)
+            .withNumStorageWriteApiStreams(2)
+            .ignoreUnknownValues()
```

Comment:
qq: is `ignoreUnknownValues` required to use missing value interpretation? or is it just added here so we can write the first row with "defaultliteral" missing from the schema?

Reply:
the latter. we are testing overriding the default with an explicit value, which requires us to know the schema. This is mostly so I don't have to create a separate input set for the tests where the input is not know.